src/lib/s2k.c

```diff
@@ -85,7 +85,7 @@ pgp_s2k_round_iterations(size_t iterations)
 uint8_t
 pgp_s2k_encode_iterations(size_t iterations)
 {
-    for (uint8_t c = 0; c <= 255; ++c) {
+    for (uint8_t c = 0; c < 255; ++c) {
```

Comment:
The section @catap quoted is actually in 3.7.2.1 in RFC 4880.

The further description is:
> Therefore, preceding the secret data there will be one of these possibilities:
>        0:           secret data is unencrypted (no passphrase)
>        255 or 254:  followed by algorithm octet and S2K specifier
>        Cipher alg:  use Simple S2K algorithm using MD5 hash

More documentation here is needed anyway 👍